### PR TITLE
Set current entry

### PR DESCRIPTION
**Pull request by alan-wu**
Add a new method to programmatically set the current view. Refactor some aspects of the viewer to separate out the initial state and available species information from the vue codes. This implementation is required to support the feature request in [this ticket](https://www.wrike.com/open.htm?id=970444593).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -116,7 +148,10 @@ export default {
     }
   },
   mounted: function() {
-    this._mapSettings = [];
+    //this.setMultiFlatmap();
```
Comment (Tehsurfer) [inline src/App.vue:151]: Still need these lines?

File: src/components/viewers/MultiFlatmap.vue [hunk 2]
```diff
@@ -140,6 +139,15 @@ export default {
         }
       }
     },
+    displayTooltip: function(info) {
+      let name = info.name;
```
Comment (Tehsurfer) [inline src/components/viewers/MultiFlatmap.vue:143]: Is info always defined? Would throw an error if not

File: src/components/MapContent.vue [hunk 3]
```diff
@@ -91,6 +92,69 @@ export default {
...
+          //          using this
+
+          //Look for the key in the available species array
+          const key = findSpeciesKey(state);
+          if (key) {
+            const currentState = this.getState();
+            if (currentState && currentState.entries) {
+              for (let i = 0; i < currentState.entries.length; i++) {
+                const entry =  currentState.entries[i];
+                if (entry.type === "MultiFlatmap") {
```
Comment (Tehsurfer) [inline src/components/MapContent.vue:134]: Do we only set the state for multiflatmaps?

File: src/components/scripts/utilities.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+/*
+ * Look for the key in availableSpecies with the provided condition
+ */
+const findSpeciesKey = condition => {
+  if (condition) {
+    const list = availableSpecies();
+    const keys = Object.keys(list);
+    for (let i = 0; i < keys.length; i++) {
```
Comment (Tehsurfer) [inline src/components/scripts/utilities.js:56]: Would it be cleaner to use:

```
for(let key in list){
    // then use 'key' instead of keys[i] 
 }
 ```

**Review by Tehsurfer — APPROVED** [review]
I like the changes! I put a few small comments

File: src/App.vue (hunk 1 above)
Comment (alan-wu) [reply src/App.vue:151]: Optional but I keep it there for testing before the map is properly loaded.

File: src/components/MapContent.vue (hunk 3 above)
Comment (alan-wu) [reply src/components/MapContent.vue:134]: Yes, this is for setting the multiflatmap and there is only one entry of multiflatmap.

File: src/components/scripts/utilities.js (hunk 4 above)
Comment (alan-wu) [reply src/components/scripts/utilities.js:56]: Good idea.

File: src/components/viewers/MultiFlatmap.vue (hunk 2 above)
Comment (alan-wu) [reply src/components/viewers/MultiFlatmap.vue:143]: True.

File: src/App.vue (hunk 1 above)
Comment (Tehsurfer) [reply src/App.vue:151]: Yeah good idea

**Pull request merged**